tips/TIP-0048/openapi3-core.yaml

```
@@ -0,0 +1,3718 @@
+      responses:
+        '200':
+          description: "Successful operation."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CongestionResponse'
+              examples:
+                CongestionResponse:
+                  $ref: '#/components/examples/get-congestion-estimate-response-example'
+        '400':
+          description: "Unsuccessful operation: indicates that the provided data is invalid."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BadRequestResponse'
+        '403':
+          description: "Unsuccessful operation: indicates that the endpoint is not available for public use."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ForbiddenResponse'
+        '500':
+          description: "Unsuccessful operation: indicates that an unexpected, internal server error happened which prevented the node from fulfilling the request."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InternalErrorResponse'
+        '503':
+          description: "Unsuccessful operation: indicates that the node is not synced."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ServiceUnavailableResponse'
+
+  '/api/core/v3/rewards/{outputId}':
+    get:
+      tags:
+        - rewards
+      summary: Returns the totally available Mana rewards of an account or delegation output decayed up to endEpoch index provided in the response.
+      description: >-
+        Returns the totally available Mana rewards of an account or delegation output up until the current epoch index. 
+        Note that rewards for an epoch only become available at the beginning of the next epoch. 
+        If the end epoch of a staking feature is equal or greater than the current epoch, the rewards response will not include the potential future rewards for those epochs.
+        startEpoch and endEpoch indicates the actual range for which reward value is returned and decayed for.
+      parameters:
+        - in: path
+          name: outputId
+          schema:
+            type: string
+          example: "0xf532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d"
+          required: true
+          description: Output ID of an account or delegation output.
+        - in: query
+          name: slotIndex
+          schema:
+            type: integer
+          example: 50
+          description: A client can specify a slot index explicitly, which should be equal to the slot it uses as the commitment input for the claiming transaction. This parameter is only recommended to be provided when requesting rewards for a Delegation Output in delegating state (i.e. when Delegation ID is zeroed). 
```

```suggestion
          description: A client can specify a slot index explicitly, which should be equal to the slot it uses as the commitment input for the claiming transaction to ensure the node calculates the rewards identically as during transaction execution. Rewards are decayed up to the epoch corresponding to the given slotIndex + MinCommittableAge. For a Delegation Output in delegating state (i.e. when Delegation ID is zeroed), that epoch - 1 is also used as the last epoch for which rewards are fetched. Callers that do not build transactions with the returned values may omit this value in which case it defaults to the latest committed slot, which is good enough to, e.g. display estimated rewards to users.
```

Update from latest adjustments to the rewards API.